apps/emqx_message_transformation/src/emqx_message_transformation.erl

```diff
@@ -664,7 +664,7 @@ trace_failure(#{log_failure := #{level := Level}} = Transformation, Msg, Meta0)
     ?tp(message_transformation_failed, Meta#{
         log_level => Level, name => Name, action => _Action, message => Msg
     }),
-    ?TRACE(Level, ?TRACE_TAG, Msg, Meta).
+    ?SLOG_THROTTLE(Level, #{msg => Msg, name => Name, action => _Action}, Meta#{tag => ?TRACE_TAG}).
```

Comment:
This will throttle failures from _all transformations_.  Perhaps we could throttle on a per-transformation basis, similar to what we do with actions?  That is, we could use the transformation name as the log ID to namespace them.

e.g.:
https://github.com/emqx/emqx/blob/3a03a64c3114051ecdee29cc74219dbce7b44b7e/apps/emqx_resource/src/emqx_resource_buffer_worker.erl#L1142-L1151

Reply:
changed